### PR TITLE
Add load balancer for shinyserver-pro

### PR DESCRIPTION
**Pull request by xschildw**
Setup a load balancer for the new shinyPro server

### CODE REVIEW

**Comment by xschildw** [comment]
It's for ssl, it's going to appear as 'https://shiny.synapse.org' (and replace the existing server). Only one instance behind the load balancer.

**Comment by zaro0508** [comment]
I'm confused why you need SSL at all when this is a private instance?  access to the instance is only thru VPN correct?

**Comment by xschildw** [comment]
It's not private, the apps can be accessed from outside (either directly at https://shiny.synapse.org or through https://synapse.org).

**Comment by zaro0508** [comment]
are you sure?  this tells me that it's not public.  https://github.com/Sage-Bionetworks/scicomp-provisioner/blob/master/config/prod/shinyserver-pro-prod.yaml#L30

**Comment by xschildw** [comment]
The instance is private, the load balancer forwards outside https traffic to it on 3838 (i.e. shiny.synapse.org will point to the load balancer, the load balancer sends to instance, we can ssh to instance from VPN).

File: config/prod/shinyserver-pro-prod-alb-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+parameters:
+  # The Sage deparment (Platform, CompOnc, SysBio, Governance, etc..)
+  Department: "Platform"
+  # The Sage project (Infrastructure, amp-ad, ntap, dream, etc..)
+  Project: "Infrastucture"
+  # The resource owner
+  OwnerEmail: "x.schildwachter@sagebase.org"
+
+  # load balancer configs
+  Ec2InstanceId: "i-0a41a3cbb9de74e9f"
```
Comment (zaro0508) [inline config/prod/shinyserver-pro-prod-alb-v2.yaml:13]: i would suggest you directly reference from the output of shinyserver-pro-prod stack like so..
```
Ec2InstanceId: !stack_output_external shinyserver-pro-prod::Ec2InstanceId
```

File: config/prod/shinyserver-pro-prod-alb-v2.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  SSLCertificateIdArn: "arn:aws:acm:us-east-1:055273631518:certificate/992c2d49-579a-465e-beff-1a08c284f7db"
+  VpcId: "vpc-e66bd19d"   # computevpc
```
Comment (zaro0508) [inline config/prod/shinyserver-pro-prod-alb-v2.yaml:15]: same here, reference from vpc.yaml output.. 
```
VpcId: !stack_output_external computevpc::VpcId
```

File: config/prod/shinyserver-pro-prod-alb-v2.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  AppPort: "3838"          # shiny port
+  Subnets: "subnet-21a68d7c,subnet-4f3e2b2b,subnet-77d1e458"  # public subnets
```
Comment (zaro0508) [inline config/prod/shinyserver-pro-prod-alb-v2.yaml:17]: same here:
```
Subnets:
  - !stack_output_external computevpc::PublicSubnet
  - !stack_output_external computevpc::PublicSubnet1
  - !stack_output_external computevpc::PublicSubnet2
```

File: config/prod/shinyserver-pro-prod-alb-v2.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  # The Sage deparment (Platform, CompOnc, SysBio, Governance, etc..)
+  Department: "Platform"
+  # The Sage project (Infrastructure, amp-ad, ntap, dream, etc..)
+  Project: "Infrastucture"
+  # The resource owner
+  OwnerEmail: "x.schildwachter@sagebase.org"
+
+  # load balancer configs
+  Ec2InstanceId: "i-0a41a3cbb9de74e9f"
+  SSLCertificateIdArn: "arn:aws:acm:us-east-1:055273631518:certificate/992c2d49-579a-465e-beff-1a08c284f7db"
```
Comment (zaro0508) [inline config/prod/shinyserver-pro-prod-alb-v2.yaml:14]: maybe there's a cert you can reference using !stack_output_external as well?

File: config/prod/shinyserver-pro-prod-alb-v2.yaml (hunk 1 above)
Comment (xschildw) [reply config/prod/shinyserver-pro-prod-alb-v2.yaml:13]: Seen this, will do!

File: config/prod/shinyserver-pro-prod-alb-v2.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Provision a load balancer for shiny.synapse.org
+template_path: remote/managed-alb-https-v2.yaml
+stack_name: shinyserver-pro-prod-alb-v2
```
Comment (zaro0508) [inline config/prod/shinyserver-pro-prod-alb-v2.yaml:3]: you'll also need to add a `dependencies:` section and list all the stacks you reference parameters from.

File: config/prod/shinyserver-pro-prod-alb-v2.yaml (hunk 4 above)
Comment (zaro0508) [reply config/prod/shinyserver-pro-prod-alb-v2.yaml:17]: i guess that's a no?

Comment (xschildw) [reply config/prod/shinyserver-pro-prod-alb-v2.yaml:17]: No, there's one for *.sagebase.org but not for *.synapse.org

**Pull request merged**